src/SWP/Bundle/ContentBundle/EventListener/ArticleMetadataSubscriber.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\EventListener;
+
+use SWP\Bundle\ContentBundle\ArticleEvents;
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Component\Bridge\Model\PackageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ArticleMetadataSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ArticleEvents::PRE_CREATE => 'populateMetadata',
+        ];
+    }
+
+    /**
+     * @param ArticleEvent $event
+     */
+    public function populateMetadata(ArticleEvent $event)
+    {
+        $package = $this->getPackage($event);
+        $article = $event->getArticle();
+
+        $article->setMetadata([
```

Comment:
If all the data is in the package, and persisted, wouldn't it make more sense to keep a reference in the article to the package, and retrieve the data from the package, instead of copying it? I realise that doctrine won't support relations between entities which aren't persisted in the same way, but even so, copying seems like a pretty inelegant solution.